x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapper.java

```diff
@@ -0,0 +1,448 @@
+                                             CopyTo copyTo, EnumMap<Metric, NumberFieldMapper> metricFieldMappers) {
+        super(simpleName, fieldType, defaultFieldType, indexSettings, multiFields, copyTo);
+        this.ignoreMalformed = ignoreMalformed;
+        this.metrics = metrics;
+        this.defaultMetric = defaultMetric;
+        this.metricFieldMappers = metricFieldMappers;
+    }
+
+    @Override
+    public AggregateDoubleMetricFieldType fieldType() {
+        return (AggregateDoubleMetricFieldType) super.fieldType();
+    }
+
+    @Override
+    protected String contentType() {
+        return fieldType.typeName();
+    }
+
+    @Override
+    protected AggregateDoubleMetricFieldMapper clone() {
+        return (AggregateDoubleMetricFieldMapper) super.clone();
+    }
+
+    @Override
+    protected void parseCreateField(ParseContext context, List<IndexableField> fields) throws IOException {
+        if (context.externalValueSet() == true) {
+            throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() + "] can't be used in multi-fields");
+        }
+
+        context.path().add(simpleName());
+        XContentParser.Token token = null;
+        XContentSubParser subParser = null;
+
+        try {
+            token = context.parser().currentToken();
+            if (token == XContentParser.Token.VALUE_NULL) {
+                context.path().remove();
+                return;
+            }
+
+            ensureExpectedToken(XContentParser.Token.START_OBJECT, token, context.parser()::getTokenLocation);
+            subParser = new XContentSubParser(context.parser());
+            token = subParser.nextToken();
+            while (token != XContentParser.Token.END_OBJECT) {
+                // should be an object subfield with name a metric name
+                ensureExpectedToken(XContentParser.Token.FIELD_NAME, token, subParser::getTokenLocation);
+                String fieldName = subParser.currentName();
+                Metric metric = Metric.valueOf(fieldName);
+
+                if (metrics.value().contains(metric) == false) {
+                    throw new IllegalArgumentException("Aggregate metric [" + metric +
+                        "] does not exist in the mapping of field [" + fieldType.name() + "]");
+                }
+
+                token = subParser.nextToken();
+                // Make sure that the value is a number. Probably this will change when
+                // new aggregate metric types are added (histogram, cardinality etc)
+                ensureExpectedToken(XContentParser.Token.VALUE_NUMBER, token, subParser::getTokenLocation);
+                NumberFieldMapper delegateFieldMapper = metricFieldMappers.get(metric);
+                delegateFieldMapper.parse(context);
+
+                if (Metric.value_count == metric) {
+                    Number n = context.doc().getField(delegateFieldMapper.simpleName()).numericValue();
```

Comment:
```suggestion
                    Number n = context.doc().getField(delegateFieldMapper.fieldType().name())).numericValue();
```
You need the full name to access the field.

Reply:
Fixed